frontend/src/views/admin/matching/styles.css

```diff
@@ -84,55 +73,67 @@
 }
 
 .position-list {
-    overflow-y: scroll;
+    overflow-y: auto;
+    padding-left: 10px;
+    padding-right: 10px;
+}
+
+.position-list-inner {
+    display: flex;
+    flex-direction: column;
+    row-gap: 5px;
+}
+
+/*
+ * We use button elements for position-row items so that
+ * keyboard navigation works. However, we want to completely customize
+ * the style of these elements, so we remove all the default button styling.
+ */
+:where(button.position-row) {
+    all: unset;
+}
+
+.position-row:focus {
+    outline: auto;
+}
+.position-row:hover {
+    background-color: #ddd;
 }
 
 .position-row {
     cursor: pointer;
+    position: relative;
     display: flex;
-    align-items: center;
-    height: 35px;
-    margin-right: 15px;
-    margin-bottom: 5px;
     box-shadow: 3px 2px 5px -3px #7d7d7d;
-    transition: height 0.4s ease-out;
+    transition: height 0.4s ease-out, flex-grow 0.4s ease-out;
```

Comment:
No, the transitions don't work right. Setting an explicit size is one solution, but it is a bad solution since sometimes positions have long names and a fixed size won't wrap correctly.

The best solution is probably a React component wrapper that monitors the size of the inner content and then dynamically sets the height property.